File: beecrowd_exercicios_em_Ruby/mathematics/factorial_again.rb
def fat(x)
  rslt = 1
  (2..x).each { |z| rslt *= z }
  rslt
end

n = gets.to_i

while n != 0

  rslt = 0

  (0...(Math.log(n)/Math.log(10) + 1).to_i).each do |i|
    rslt += n%10*fat(i + 1)
    n /= 10

  end

  puts rslt

  n = gets.to_i

end

File: beecrowd_exercicios_em_python/mathematics/factorial_again.py
import math


def fat(x):
    rslt = 1

    for i in range(2, x + 1):
        rslt *= i

    return rslt


n = int(input())

while n != 0:

    rslt = 0

    for i in range(1, int(math.log10(n)) + 2):
        rslt += n % 10 * fat(i)
        n //= 10

    print(rslt)

    n = int(input())

